src/actions/connectionKeyPairActions.js

```diff
@@ -174,30 +217,40 @@ export const updateConnectionKeyPairs = (mnemonic: string, privateKey: string, w
       return Promise.resolve(false);
     }
 
-    dispatch({
+    const { currentConnectionsCount, oldConnectionsCount } = numberOfConnections;
+    const totalConnections = currentConnectionsCount + oldConnectionsCount;
+    await dispatch({
       type: UPDATE_WALLET_STATE,
       payload: GENERATING_CONNECTIONS,
     });
-
-    const { currentConnectionsCount, oldConnectionsCount } = numberOfConnections;
-    const totalConnections = currentConnectionsCount + oldConnectionsCount;
-    const newKeyPairs =
-      await generateKeyPairThreadPool(
-        mnemonic,
-        privateKey,
-        totalConnections,
-        connectionKeyPairs.length,
-        lastConnectionKeyIndex);
-    const resultConnectionKeys = connectionKeyPairs.concat(newKeyPairs);
+    try {
+      const newKeyPairs =
+        await generateKeyPairThreadPool(
+          mnemonic,
+          privateKey,
+          totalConnections,
+          connectionKeyPairs.length,
+          lastConnectionKeyIndex);
+      const resultConnectionKeys = connectionKeyPairs.concat(newKeyPairs);
+      await dispatch({
+        type: UPDATE_CONNECTION_KEY_PAIRS,
+        payload: resultConnectionKeys,
+      });
+      await dispatch(saveDbAction('connectionKeyPairs', { connectionKeyPairs: resultConnectionKeys }, true));
+    } catch (e) {
+      await dispatch({
+        type: UPDATE_WALLET_STATE,
+        payload: DECRYPTING,
+      });
```

Comment:
Just to make sure, is this the correct event we want to dispatch? I'd have expected it to be an error 🤔

Reply:
Sentry is declared after the call so in a future improvement it can be used here.